test/e2e/azure_logcollector.go

```diff
@@ -50,40 +52,61 @@ func (k AzureLogCollector) CollectMachineLog(ctx context.Context, managementClus
 		return err
 	}
 
-	if err := collectLogsFromNode(ctx, managementClusterClient, m, am, outputPath); err != nil {
+	cluster, err := util.GetClusterFromMetadata(ctx, managementClusterClient, m.ObjectMeta)
+	if err != nil {
+		return err
+	}
+
+	hostname := getHostname(m, isAzureMachineWindows(am))
+
+	if err := collectLogsFromNode(ctx, managementClusterClient, cluster, hostname, isAzureMachineWindows(am), outputPath); err != nil {
 		errors = append(errors, err)
 	}
 
-	if err := collectBootLog(ctx, am, outputPath); err != nil {
+	if err := collectVMBootLog(ctx, am, outputPath); err != nil {
 		errors = append(errors, err)
 	}
 
 	return kinderrors.NewAggregate(errors)
-
 }
 
-// collectLogsFromNode collects logs from various sources by ssh'ing into the node
-func collectLogsFromNode(ctx context.Context, managementClusterClient client.Client, m *clusterv1.Machine, am *v1alpha4.AzureMachine, outputPath string) error {
-	cluster, err := util.GetClusterFromMetadata(ctx, managementClusterClient, m.ObjectMeta)
+// CollectMachinePoolLog collects logs from a machine pool.
+func (k AzureLogCollector) CollectMachinePoolLog(ctx context.Context, managementClusterClient client.Client, mp *expv1.MachinePool, outputPath string) error {
+	var errors []error
+
+	am, err := getAzureMachinePool(ctx, managementClusterClient, mp)
 	if err != nil {
 		return err
 	}
 
-	Logf("INFO: Collecting logs for machine %s in cluster %s in namespace %s\n", m.GetName(), cluster.Name, cluster.Namespace)
-	isWindows := isNodeWindows(am)
+	cluster, err := util.GetClusterFromMetadata(ctx, managementClusterClient, mp.ObjectMeta)
+	if err != nil {
+		return err
+	}
 
-	controlPlaneEndpoint := cluster.Spec.ControlPlaneEndpoint.Host
-	hostname := m.Spec.InfrastructureRef.Name
-	if isWindows {
-		// Windows host name ends up being different than the infra machine name
-		// due to Windows name limitations in Azure so use ipaddress instead
-		if len(m.Status.Addresses) > 0 {
-			hostname = m.Status.Addresses[0].Address
-		} else {
-			Logf("INFO: Unable to collect logs as node doesn't have addresses")
+	isWindows := isAzureMachinePoolWindows(am)
+
+	for i, instance := range mp.Spec.ProviderIDList {
+		hostname := mp.Status.NodeRefs[i].Name
```

Comment:
For windows this might need to use `gethostname()` func you added?

Reply:
not for machine pools (I'm using the node name directly), seems to be working: https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/1389/pull-cluster-api-provider-azure-e2e-windows/1398063792918630400/artifacts/clusters/capz-e2e-aq3vvw/machine-pools/capz-e2e-aq3vvw-mp-win/win-p-win000000/